share/functions/fish_commandline_prepend.fish

```diff
@@ -3,14 +3,24 @@ function fish_commandline_prepend --description "Prepend the given string to the
         commandline -r $history[1]
     end
 
-    set -l escaped (string escape --style=regex -- $argv[1])
     set -l process (commandline -p | string collect)
-    if set process (string replace -r -- "^$escaped " "" $process)
+
+    set -l to_prepend "$argv[1] "
+    if string match -qr '^ ' "$process"
+        set to_prepend " $argv[1]"
+    end
+
+    set -l length_diff (string length -- "$to_prepend")
+    set -l cursor_location (commandline -pC)
+
+    set -l escaped "$(string escape --style=regex -- $to_prepend)"
```

Comment:
Now that I'm looking at it again I think this might be better:
```fish
set -l escaped (string escape --style=regex -- "$to_prepend")
```

Let me know if I should change it, or if it doesn't matter.

Reply:
It doesn't really matter in this context - `$(...)` only makes a difference if you're trying to avoid splitting multiple lines into an array and want them as a single string. However the convention in our codebase would be to not use a quoted subshell. You also don't need the quotes in `"$to_prepend"` in your revised alternative either, as all fish variables are implicitly quoted when passed as parameters. Isn't fish just too easy?

Reply:
> You also don't need the quotes in `"$to_prepend"`

Right you are, that does make things easier. Thanks!